src/agent/src/rpc.rs

```diff
@@ -760,6 +787,45 @@ impl AgentService {
 
         Ok(())
     }
+
+    #[cfg(feature = "sealed-secret")]
+    async fn unseal_secret_async(&self, sealed: &str) -> Result<sealed_secret::UnSealSecretOutput> {
```

Comment:
I'm not sure about this:

as there are functions `unseal_secret_async()`, `remove_if_sock_exist()`, etc are related to sealed secret feature. Is it possible to put them into a separate file or a `mod` block. In this way we might only need one `#[cfg(feature = "sealed-secret")]` flag to enable the file or the `mod` block?

Reply:
move related code to src/agent/src/cdh.rs